pennylane/tape/tapes/cv_param_shift.py

```diff
@@ -368,27 +378,48 @@ def parameter_shift(self, idx, device, params, **options):
             1st order method may be more efficient unless it's really easy to
             experimentally measure arbitrary 2nd order observables.
 
+        .. warning::
+
+            The 2nd order method can only be executed on devices that support the
+            :class:`~.PolyXP` observable.
+
         Args:
             idx (int): trainable parameter index to differentiate with respect to
-            device (.Device): a PennyLane device that can execute quantum operations and return
-                measurement statistics
             params (list[Any]): the quantum tape operation parameters
 
         Keyword Args:
             force_order2 (bool): iff True, use the order-2 method even if not necessary
+            device (.Device): A PennyLane device that can execute quantum operations and return
+                measurement statistics. This keyword argument is required, as the device labels
+                are required to generate the quantum tapes for computing the gradient.

```

Comment:
It seems like this is "required" because we need to not only know the wires, but also whether it supports `PolyXP`

Reply:
Yep, I don't like this, but I can't think of another solution. Removing the device execution from the gradient methods was very smooth everywhere, except the CV methods, where it required the device for (a) wire information, and (b) to check if PolyXP was supported.

Reply:
It would be nice if there was a way for it to tell the `tape.jacobian` method 'hey, the device doesn't support PolyXP? Then don't call me'

Reply:
Perhaps it would be nice if the gradient methods could declare in advance what operations they support. Similar to devices. This would also help with the qubit parameter shift and CR gates

Reply:
on this particular comment, I was also calling out that the docstring was not 100% accurate